### DISCUSSION TITLE
Function Keys (ie F1, F2, etc) support in FocusScope

### DISCUSSION THREAD

**Discussion by Vadoola** [slint-ui/slint#4401] [Q&A] [answered]
I've been working on adding Global Hot Key Support to [Tomotroid](https://github.com/Vadoola/Tomotroid). The basic support in the code has been there for a while now, just with hard coded hot keys. The default Hot Keys used were Ctrl+F1, Ctrl+F2, and Ctrl+F3.

I'm now working on the UI part so the user can customize these global hotkeys. I have it set up using a Focus Scope so I could capture the keys pressed. However, I can't seem to capture which function key is pressed. I've tested this on both Windows 11 and Linux, and if I use any Function key the text value is always xEF (for some reason in VSCodium on Windows the debugger shows xEF, and on Linux it shows 239, either way same value). Meaning there is no way for me to determine which Function Key is pressed (or released)

Is this possible with FocusScope at the moment? If not any thoughts on how I could work around this?
I did see an Issue or Discussion a couple of days ago (I'm not seeing it right now) where it was mentioned that a new (or recent) version of Winit has reworked KeyEvent system. Is this something I might need to wait on an update to get working?

Edit: I found the discussion I was thinking about: https://github.com/slint-ui/slint/discussions/3189
I guess it looks like support for scan codes is what I would be after if I understand correctly.  I see @hunger said "We should be able to benefit from the extra information they provide and should be able to improve our key events accordingly." Is there any tracking issue for this or plans for how it might work?

**Comment by ogoffart** [comment, marked as answer]
The function key are supported in the FocusScope. The text is `Key.F1`, `Key.F2`, ...  (documented there  https://slint.dev/releases/1.3.2/docs/slint/src/language/builtins/namespaces#key )
Also in Rust the special key codes are accessible trouch the Key enum: https://docs.rs/slint/latest/slint/platform/enum.Key.html which is convertible to char or string
Ctrl should work the same way. Unless the keys is already handled by the window manager.

**Reply by Vadoola** [reply]
Hmm....I hadn't actually tried matching against the Key enum yet, because the event.text always seemed to have the exact same value no matter which function key was pressed, so I just assumed it wasn't quite working right. I did a very quick test, and it does seem to differentiate between F1 and F2, so clearly it's working. I know the debuggers don't always work the best with Rust, and I've seen other odd issues in the past, I wonder if the debugger was showing me the wrong info or something.

Thanks for the help.
